src/client/dfs/dfs.c

```diff
@@ -4888,7 +4889,11 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 		rc = daos_array_set_size(obj->oh, th, stbuf->st_size, NULL);
 		if (rc)
 			D_GOTO(out_obj, rc = daos_der2errno(rc));
-		rstat.st_size = stbuf->st_size;
+
+		/* If the size was set then do a second stat to fetch the new mtime */
+		rc = entry_stat(dfs, th, oh, obj->name, len, obj, &rstat, &obj_hlc);
```

Comment:
the extra size query can be expensive and should be a no go. also here we only need fetch the mtime, not the entire inode entry. I'd rather we do this in this PR in addition to doing the improvement you suggested and port to 2.2 rather than do separate PRs.
- for the first stat above, we can add a bool to entry_stat to not fetch the size if this is a size set.
- for this stat, we just need a fetch of the mtime (sec and nsec)

Reply:
let me know if you want me to do those changes on this PR, and I would be happy to push that.

Reply:
Please do.  The first would be trivial but I've not looked into how the mtime fetch works.

Reply:
See https://github.com/daos-stack/daos/pull/10093
i realized that for the second bullet we do need a daos_array_stat to fetch the max epoch since we are not updating the mtime in the inode entry. So overall my PR removes 1 extra array_stat, and an entry fetch in case of set_size.

Reply:
NLT seems to have failed there, but im not sure why. need to stop today. @ashleypittman please check